test/com/google/javascript/jscomp/ConvertIIFEArgsToVarsTest.java

```diff
@@ -0,0 +1,100 @@
+package com.google.javascript.jscomp;
+
+/**
+ * Created by chad on 10/25/16.
+ */
+public final class ConvertIIFEArgsToVarsTest extends CompilerTestCase {
+  @Override
+  public CompilerPass getProcessor(Compiler compiler) {
+    return new ConvertIIFEArgsToVars(compiler);
+  }
+
+  @Override
+  public void setUp() {
+    enableNormalize();
+  }
+
+  public void testBasic() {
+    test(
+        LINE_JOINER.join(
+            "var a = 1;",
+            "var b = 2;",
+            "(function(c, d) {",
+            "  console.log(c, d);",
+            "})(a, b)"),
+        LINE_JOINER.join(
+            "var a = 1;",
+            "var b = 2;",
+            "(function() {",
+            "  var c = a;",
+            "  var d = b;",
+            "  console.log(c, d);",
+            "})()"));
+  }
+
+  public void testExplicitThis() {
+    test(
+        LINE_JOINER.join(
+            "var a = 1;",
+            "var b = 2;",
+            "(function(c, d) {",
+            "  console.log(c, d);",
+            "}).call(null, a, b)"),
+        LINE_JOINER.join(
+            "var a = 1;",
+            "var b = 2;",
+            "(function() {",
+            "  var c = a;",
+            "  var d = b;",
+            "  console.log(c, d);",
+            "}).call(null)"));
+  }
+
+  public void testNested() {
+    test(
+        LINE_JOINER.join(
+            "var a = 1;",
+            "var b = 2;",
+            "(function(c, d) {",
+            "  (function(e, f) {",
+            "    console.log(e, f);",
+            "  })(c, d);",
+            "}).call(null, a, b);"),
+        LINE_JOINER.join(
+            "var a = 1;",
+            "var b = 2;",
+            "(function() {",
+            "  var c = a;",
+            "  var d = b;",
+            "  (function() {",
+            "    var e = c;",
+            "    var f = d;",
+            "    console.log(e, f);",
+            "  })();",
+            "}).call(null)"));
+  }
+
+  public void testArgumentsReference() {
+    testSame(LINE_JOINER.join(
+        "var a = 1;",
+        "var b = 2;",
+        "(function(c, d) {",
+        "  var len = arguments.length;",
+        "  console.log(c, d, len);",
+        "}).call(null, a, b);"));
+  }
+
+  public void testUndefinedAlias() {
+    test(
+        LINE_JOINER.join(
+            "(function(c, d) {",
+            "  console.log(c, d);",
+            "})();"),
+        LINE_JOINER.join(
+            "(function() {",
+            "  var c = undefined;",
+            "  var d = undefined;",
+            "  console.log(c, d);",
+            "})();"));
+  }
+}
```

Comment:
Can you add a test where the arguments are more than the formal parameters? I wanna make sure we don't crash in that case.